sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -335,6 +336,32 @@ abstract class ExternalCatalog
     postToAll(RenameFunctionEvent(db, oldName, newName))
   }
 
+  final def checkSchemaContainsBucketingColumns(
```

Comment:
But then `InMemoryCatalog.alterTableSchema` will not have this check. That makes few tests to fail. Dont we want to have consistency wrt this check in both `InMemoryCatalog` and  `SessionCatalog` ?